src/reporters/http_sender.js

```diff
@@ -107,11 +107,15 @@ export default class HTTPSender {
       return;
     }
 
-    this._reset();
-
     const requester = this._url.protocol === 'https:' ? https.request : http.request;
 
     const req = requester(this._httpOptions, resp => {
+      
+      if(resp.statusCode && (resp.statusCode >= 200 || resp.statusCode < 227))
+      {
+        this._reset();  
+      }
```

Comment:
I dont think this is correct behavior. First, if we dont clear the buffer then the same spans might be submitted again due to concurrency if this request gets stuck. Second, if you want to build in error handling then the code can not proceed as usual and report success metrics.